src/validator.ts

```diff
@@ -30,10 +37,54 @@ export = function validator(definition: StateMachine, opts?: ValidationOptions):
         errors.push(...duplicateStateNames(definition, options));
         errors.push(...missingTerminalStateErrors(definition, options));
         errors.push(...mustNotHaveDuplicateFieldNamesAfterEvaluation(definition, options))
-        errors.push(...terminalStateWithNext(definition, options));
-        errors.push(...waitDuration(definition, options));
-        errors.push(...taskChecks(definition, options));
-        errors.push(...mapChecks(definition, options));
+        errors.push(...stateChecks(definition, options, [
+            {
+                filter: IsTask,
+                checker: AtMostOne({
+                    props: ["TimeoutSeconds", "TimeoutSecondsPath"],
+                    errorCode: StateMachineErrorCode.TaskTimeoutError})
+            },
```

Comment:
removed the separate functions in favor of inlined logic here for the state checks

note that the `filter` and `checker` params are both `functions`. The code here only uses the helper functions `AtMostOne` and `ExactlyOne` but custom filters/checkers are possible